docs/userstories/v1/egress-control-in-anp/egress-control-outside-cluster.md

```diff
@@ -0,0 +1,56 @@
+
+### Summary
+
+1. **As a** cluster administrator **I want** to restrict traffic from
+specific cluster workloads to all or specific destinations outside the
+cluster **so that** I can enforce security for south-north traffic.
+
+2. **As a** cluster administrator **I want** to explicitly allow traffic
+from cluster workloads to handpicked destinations outside the cluster and
+deny the rest **so that** I can guarantee cluster wide egress traffic
+guardrails irrespective of whatever network policies are defined by the
+developer for specific namespaces.
+
+### Description
+
+Often times cluster administrators want to explicitly control the
+egress traffic from workloads towards external destinations like allowing
+some (not all) workloads to access a secure service outside the cluster.
+
+They might also want to setup cluster wide safeguard rules for egress in
+general so that they don't have to worry about when namespace specific
+policies are created by users at any given point.
+
+### Acceptance Criteria
+
+1. If I add an admin network policy to deny traffic from a specific "subject"
+of workloads towards an external destination (optionally provide port
+restrictions) then those workloads should not be able to connect to that
+external destination.
+
+2. If I apply an admin network policy with lower priority egress rules to deny
+traffic from a specific "subject" of workloads towards all external
+destinations and then apply higher priority egress rules to allow traffic for
+those workloads towards well defined destinations then those workloads should
+not be able to connect to any other external destinations than the explicitly
+allowed ones.
+
+[
+Also, here are a few points that need to be addressed:
+
+1. Should we also address Ingress case for north-south traffic? Do we have
+requests from users to restrict access INTO the cluster workloads from external
+entities? I'd think not since clusterCIDRs are not generally routable
+from outside anyways.
```

Comment:
Ingress is a lot harder to deal with so the plan had been to ignore it for now

Reply:
agreed. I'll go with egress only for now.